### PR TITLE
Fix #4131, #4130, #4129 - Fixes Duplicate items in playlist. Fixes control-center not showing playback info properly.

### LINKED ISSUE #4102: Playlist: no way to play video in the background with a different speed
### Description

Brave iOS Playlist resets the video speed to 1x when playing in the background. There's no way to play video other than on 1x when the screen is locked.

### Steps to Reproduce 

  1. Start playing a video in the Playlist.
  2. Change the speed to 2x.
  3. Lock the screen. (Video is paused.)
  4. Start the video again from control on the lock screen.

**Actual result:**

The video is played (in the background) with 1x speed.

**Expected result:**

The video should be played with 2x speed.

**Reproduces how often:**

Every time.

**Brave Version:** 

1.30.1 (21.8.17.19)

**Device details:** 

iPhone 11 (iOS 13.6)

### Additional Information

Possibly related: https://community.brave.com/t/playlist-keeps-resetting-speed-on-play-wont-persist-across-videos-or-after-pause/269748

### LINKED ISSUE #4129: Playlist continues to play even after removing the playlist item
### Description: 
Playlist continues to play even after removing the playlist item

### Steps to Reproduce 
  1. Visit a page that has embedded videos like twitter
  2. Add the same video twice once via main twitter page and once via the individual tweet page
  3. Open playlist ensure there are two copies of videos listed
  4. Play the first video on the list of the two
  5. Swipe delete the second video, both get deleted but video continues to play

**Actual result:** 

https://user-images.githubusercontent.com/17010094/131988749-51d223c6-a4e4-4ebd-88b0-cde22e418d7d.MP4

**Expected result:**
Stop video when playlist item is deleted

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 
1.31(21.9.2.11)

**Device details:** 
All

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)? 

### Additional Information 
Issue repro's on 1.30.1 as well so its an existing bug 
cc: @Brandon-T

### LINKED ISSUE #4130: Playlist re-adds videos to list when its removed
### Description: 
Playlist re-adds videos to list when its removed

### Steps to Reproduce 
  1. Open two YT videos on two tabs
  2. Add first tab video to Playlist via URL bar button
  3. Open playlist and ensure video is added
  4. Close playlist and switch to second YT tab
  5. Add the video to playlist again via URL bar button
  6. Open playlist
  7. Swipe delete the list items, both get re-added

**Actual result:** 
https://user-images.githubusercontent.com/17010094/131991061-4fce4545-7fd7-4c1c-8846-6649cbe4c87f.MP4

**Expected result:**
Should not re-add videos back to playlist when its removed

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Repro'd once 

**Brave Version:** 
1.31(21.9.1.16)

**Device details:** 
iPhone 7+ (iOS 14.5.1)

**Website problems only:**
- did you check with Brave Shields down? NA
- did you check in Safari/Firefox (WkWebView-based browsers)? NA

### Additional Information
cc: @Brandon-T

### LINKED ISSUE #4131: Deleted video still shows as added to playlist and shows open in playlist
### Description: 
Deleted video still shows as added to playlist and shows open in playlist

### Steps to Reproduce 
  1. Open two YT videos in individual tabs
  2. Play video on tab 1 and add it to playlist
  3. Switch to video on tab 2 and add it to playlist
  4. Open playlist and delete both videos that were added
  5. Close playlist and go back to tab2, shows Add to playlist button as expected
  6. Switch to tab 1, shows video is added to playlist
  7. Opening the button shows open in playlist. When playlist is opened its empty coz we deleted the video in step 4

**Actual result:** 
https://user-images.githubusercontent.com/17010094/131991061-4fce4545-7fd7-4c1c-8846-6649cbe4c87f.MP4

**Expected result:**
Update button for backgrounded tabs when playlist item is removed 

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 
1.31(21.9.1.16)

**Device details:** 
iPhone 7+ (iOS 14.5.1)

**Website problems only:**
- did you check with Brave Shields down? NA
- did you check in Safari/Firefox (WkWebView-based browsers)?  NA

### Additional Information
cc: @Brandon-T

### PR DESCRIPTION
**Pull request by Brandon-T**
## Summary of Changes
- Fixes Duplicating items when adding from separate tabs.
- Fixes LockScreen not showing the Control Center info properly.

This pull request fixes #4131, 
fixes #4130, 
fixes #4129,

Incidentally, this PR also fixes #4102 just as a side-effect of changes made to control-center.

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

STR in linked tickets

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Browser/Playlist/PlaylistMediaInfo.swift [hunk 1]
```diff
@@ -95,9 +102,9 @@ class PlaylistMediaInfo: NSObject {
         
         UIApplication.shared.beginReceivingRemoteControlEvents()
         updateNowPlayingMediaInfo()
-        rateObserver = playerView.player.observe(\AVPlayer.rate, changeHandler: { [weak self] _, _ in
-            self?.updateNowPlayingMediaInfo()
-        })
+//        rateObserver = playerView.player.observe(\AVPlayer.rate, changeHandler: { [weak self] _, _ in
+//            self?.updateNowPlayingMediaInfo()
```
Comment (iccub) [inline Client/Frontend/Browser/Playlist/PlaylistMediaInfo.swift]: can be removed?

**Review by iccub — COMMENTED** [review]
No changes on development branch required?

File: Client/Frontend/Browser/Playlist/PlaylistMediaInfo.swift (hunk 1 above)
Comment (Brandon-T) [reply Client/Frontend/Browser/Playlist/PlaylistMediaInfo.swift]: Fixed :)

**Comment by Brandon-T** [comment]
> No changes on development branch required?

Development already has its own PR which is part of the CarPlay iOS 14+ PR and does the exact same thing. I just had to remove all the car-play stuff for "Release" branch.

**Comment by iccub** [comment]
thank you for making these `release` specific PRs, i know it's a pain

**Review by soner-yuksel — APPROVED** [review]

**Pull request merged**